utils/index.d.ts

```diff
@@ -1351,11 +1351,16 @@ export interface IAddUserAgent {
  */
 export declare function appendExtensionUserAgent(existingUserAgent?: string): string;
 
+export type AzureExtensionApiFactory<T extends AzureExtensionApi = AzureExtensionApi> = {
+    apiVersion: string,
+    createApi: (options?: GetApiOptions) => T
+};
+
 /**
  * Wraps an Azure Extension's API in a very basic provider that adds versioning.
  * Multiple APIs with different versions can be supplied, but ideally a single backwards-compatible API is all that's necessary.
  */
-export declare function createApiProvider(azExts: AzureExtensionApi[]): AzureExtensionApiProvider;
+export declare function createApiProvider(azExts: AzureExtensionApiFactory[]): AzureExtensionApiProvider;
```

Comment:
Is this a breaking change, i.e. would require bumping the major version?

Reply:
Yes it is. I considered trying to make it backwards compatible by accepting `(AzureExtensionApi | AzureExtensionApiFactory)[]`, but decided against it.

A lot of the extensions use `createApiProvider`, but only to expose an API with reveal. Maybe we should leave this `createApiProvider` function alone, and add a new method that we can use going forward.

The breaking change is pretty easy to adapt when needed, however I think it impacts all extensions.

Reply:
Ok I went ahead and made it backwards compatible. This will make our lives easier for a while until we can convert all APIs over to the factory pattern.